crates/hive-utils/src/madara/utils.rs

```diff
@@ -280,12 +279,12 @@ mod tests {
             let contract_account_class_hash = env.kakarot().contract_account_class_hash;
             let counter_address =
                 StarknetContractAddress(Into::<StarkFelt>::into(counter_genesis_address).try_into().unwrap());
-            let counter_storage_record = StorageRecord {
-                nonce: Nonce(StarkFelt::from(0u8)),
-                class_hash: ClassHash(contract_account_class_hash.into()),
-                storage: counter_storage,
-            };
-            starknet.storage.insert(counter_address, counter_storage_record);
+
+            starknet.set_class_hash_at(counter_address, ClassHash(contract_account_class_hash.into())).unwrap();
+            starknet.set_nonce(counter_address, Nonce(StarkFelt::from(0u8)));
```

Comment:
i thought nonce were suppose to start at 1

Reply:
Same as above, we are setting the nonce of an evm counter contract.

Reply:
@Eikix "nonce starts at 1" is not fully accurate, nonce is incremented in a deploy tx from 0 to 1 because it actually runs the deploy byteode, but afaiu in a CREATE/CREATE2 the nonce is 0 though a contract is deployed

Reply:
ok I misunderstood Elias comment, we also have to be careful because in the above I set the protocol level nonce, so if the evm nonce needs to be set we need to add this.

Reply:
for contract account it depends if it's CREATE I guess (just checking rn), so may not be an issue to leave 0

Reply:
ok just checked and event after CREATE, nonce is indeed 1

Reply:
which means that we probably have an issue here where we should increment the nonce
https://github.com/ClementWalter/kakarot/blob/bf4a98365bf9a66e40cd8f34e8c252ac913d6833/src/kakarot/instructions/system_operations.cairo#L943

Reply:
Will do